packages/firebase/index.d.ts

```diff
@@ -7906,10 +7972,24 @@ declare namespace firebase.firestore {
      */
     onSnapshot(
       options: SnapshotListenOptions,
-      onNext: (snapshot: DocumentSnapshot) => void,
+      onNext: (snapshot: DocumentSnapshot<T>) => void,
       onError?: (error: Error) => void,
       onCompletion?: () => void
     ): () => void;
+
+    /**
+     * Applies a custom data converter to this DocumentReference, allowing you
+     * to use your own custom model objects with Firestore. When you call
+     * set(), get(), etc. on the returned DocumentReference instance, it will
+     * use the provided converter to convert data to/from type U (your custom
```

Comment:
Maybe: "... returned DocumentReference instance, the provided converter will convert between Firestore data and your custom type U".